### DISCUSSION TITLE
[4.0.0]Can't delete a specific admin - Key (id) is still referenced from table "django_admin_log"

### DISCUSSION THREAD

**Discussion by CTV-2023** [netbox-community/netbox#15987] [Help Wanted!] [answered]
Hi,
After an upgrade to version 4.0.0, I don't manage to delete one specific admin (SAML user created with social_auth). An error 500 is returned. As I don't have the issue with other users, it doesn't seem to be a bug on the version, so I'll start with a simple "discussion"

> <class 'django.db.utils.IntegrityError'>
> 
> update or delete on table "users_user" violates foreign key constraint "django_admin_log_user_id_c564eba6_fk_auth_user_id" on table "django_admin_log"
> DETAIL:  Key (id)=(31) is still referenced from table "django_admin_log".
> 

Python version: 3.11.2
NetBox version: 4.0.0
Plugins: None installed

Any idea ?

**Comment by jeremystretch** [comment, marked as answer]
The error likely stems from an entry in the separate Django admin log (not to be confused with NetBox's global change log) which references the user account with ID 31. You can likely resolve the issue by deleting any of the offending records using a SQL shell `manage.py dbshell`:

```
netbox=> DELETE FROM django_admin_log WHERE id=31;
```

Or, to wipe out all the logs to mitigate additional errors:

```
netbox=> DELETE FROM django_admin_log;
```

**Reply by CTV-2023** [reply]
Thanks. As this is my test server, It's hard to understand what happened here (issue created by the update or present before the update...). Solved anyway 👍

**Comment by multco-troy** [comment]
I had an issue after the upgrade to 4.3 with the change log not showing any entries when logging in as a particular user (even with the cache disabled).  I had to delete the logs from `django_admin_log` and delete / add the user to solve the issue.

**Comment by jeremystretch** [comment]
FYI this has been addressed under #20341 in NetBox v4.4.1.
